src/app/services/dynamic-form.service.ts

```diff
@@ -0,0 +1,93 @@
+import {
+  DynamicFormInterface,
+  FieldInterface,
+  FormFieldTypeEnum,
+  OptionInterface
+} from '../shared/components/forms/dynamic-form/dynamic-form.model';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class DynamicFormService {
+  constructor(private fb: FormBuilder) {}
+
+  private FormControlResolver: any = {
+    [FormFieldTypeEnum.Checkbox]: (formItem: FieldInterface) => {
+      return formItem.options && this.fb.array(formItem.options.map(() => new FormControl(false)));
+    },
+    [FormFieldTypeEnum.Slider]: (formItem: FieldInterface) => {
+      const formGroup = this.fb.group({});
+      formItem.options &&
+        formItem.options.forEach((option: OptionInterface) => {
+          formGroup.addControl(option.label, this.fb.control(option.value));
+        });
+      return formGroup;
+    },
+    default: (formItem: FieldInterface) => {
+      const validationErrors = formItem.validators && formItem.validators.map((validator) => validator.error);
+      return this.fb.control(formItem.value || '', validationErrors);
+    }
+  };
+
+  resolveFormControl(formItem: FieldInterface): AbstractControl {
+    return this.FormControlResolver[formItem.type]
+      ? this.FormControlResolver[formItem.type](formItem)
+      : this.FormControlResolver.default(formItem);
+  }
+
+  setupFormFields(dynamicForm: FormGroup, formConfig: DynamicFormInterface): void {
+    formConfig.fields.forEach((formItem: FieldInterface) => {
+      const formControl = this.resolveFormControl(formItem);
+      dynamicForm.addControl(formItem.id, formControl);
+    });
+  }
+
+  resetForm(dynamicForm: FormGroup, formConfig: DynamicFormInterface): void {
+    dynamicForm.reset();
+    this.resetValuesForSliderType(dynamicForm, formConfig);
+  }
+
+  resetValuesForSliderType(dynamicForm: FormGroup, formConfig: DynamicFormInterface): void {
+    formConfig.fields.forEach((formItem: FieldInterface) => {
+      if (formItem.type === FormFieldTypeEnum.Slider) {
+        const id = formItem.id;
+        dynamicForm.patchValue({
+          [id]: {
+            min: formItem.min,
+            max: formItem.max
+          }
+        });
+      }
+    });
+  }
+
+  getModifiedCheckboxValues(dynamicForm: FormGroup, checkboxFields: FieldInterface[]): Object {
+    return checkboxFields.reduce((acc: {[key: string]: string[] | undefined}, checkboxField: FieldInterface) => {
+      if (checkboxField.options) {
+        const modValue = this.transformCheckboxesArray(dynamicForm, checkboxField.id, checkboxField.options);
+        acc[checkboxField.id] = modValue;
+      }
+
+      return acc;
+    }, {});
+  }
+
+  transformCheckboxesArray(dynamicForm: FormGroup, checkboxFieldId: string, options: OptionInterface[]): string[] {
+    const checkboxValues = dynamicForm.value[checkboxFieldId];
+    const selectedOptions = checkboxValues
+      .map((isChecked: boolean, index: number) => (isChecked ? options[index].label : null))
```

Comment:
```suggestion
      .map((isChecked: boolean, index: number) => (isChecked ? options[index].value: null))
```

Reply:
Ok, thanks. I changed, works the same. By the way, I have a question about the price field. Now I implemented form in the way that when I change the checkbox (at least one is checked) that it is shown in the forms value (as we talked yesterday). So if checkbox is empty - it is not added to the form fields. But what about slider? For now slider price is always in the form fields, even when user did not use it yet. Maybe it is correct because slider already have start and end values... Or maybe to show it only in a case when this field has a pristine value in false. How to do it better? To add or not slider price values if user did not change them?

Reply:
We can investigate some websites, and for example in Rozetka, this values set by default: 
![image](https://github.com/Denys1994q/online-shop/assets/89141792/18c5d173-b5c7-49ed-a034-c8f255363212)

Reply:
So, I think is the way you currently handle it is correct